nrf_security/src/mbedcrypto_glue/CMakeLists.txt

```diff
@@ -3,55 +3,66 @@
 #
 # SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
 #
-if(CONFIG_NRF_CRYPTO_GLUE_LIBRARY)
-  zephyr_include_directories(
-      ${NRF_MBEDTLS_ROOT}/include/mbedcrypto_glue/mbedtls
-  )
 
+if(CONFIG_NRF_CRYPTO_GLUE_LIBRARY)
   zephyr_library_named(mbedcrypto_glue)
-  zephyr_library_sources(
-      aes_alt.c
-      ccm_alt.c
-      dhm_alt.c
-      ecdh_alt.c
-      ecdsa_alt.c
-      platform_alt.c
-      rsa_alt.c
-  )
 
+  #
+  # Add files to glue if they are glued
+  #
+  zephyr_library_sources_ifdef(CONFIG_GLUE_MBEDTLS_AES_C    aes_alt.c)
+  zephyr_library_sources_ifdef(CONFIG_GLUE_MBEDTLS_CCM_C    ccm_alt.c)
+  zephyr_library_sources_ifdef(CONFIG_GLUE_MBEDTLS_DHM_C    dhm_alt.c)
+
+  #
+  # Add include directories shared by all glue libraries
+  #
   zephyr_library_include_directories(
-      ${NRF_MBEDTLS_ROOT}/include/mbedcrypto_glue
+      "${NRF_MBEDTLS_ROOT}/include/mbedcrypto_glue"
+      "${common_includes}"
+      "${generated_includes}"
   )
 
-  zephyr_library_compile_definitions(
-      -DMBEDTLS_AES_ALT
-      -DMBEDTLS_CCM_ALT
-      -DMBEDTLS_CMAC_ALT
-      -DMBEDTLS_DHM_ALT
-      -DMBEDTLS_ECDH_GEN_PUBLIC_ALT
-      -DMBEDTLS_ECDH_COMPUTE_SHARED_ALT
-      -DMBEDTLS_ECDSA_VERIFY_ALT
-      -DMBEDTLS_ECDSA_SIGN_ALT
-      -DMBEDTLS_ECDSA_GENKEY_ALT
-      -DMBEDTLS_GCM_ALT
-      -DMBEDTLS_RSA_ALT
-  )
+  #
+  # Add mbed TLS config file shared by all glue libraries
+  #
+  zephyr_library_compile_definitions(-DMBEDTLS_CONFIG_FILE="${CONFIG_MBEDTLS_CFG_FILE}")
+
+  #
+  # Create glue libraries
+  #
+  if (CONFIG_CC310_BACKEND)
+    include(${CMAKE_CURRENT_LIST_DIR}/cc310/CMakeLists.txt)
+  endif()
```

Comment:
Files that are **add_subdirectory'd** should be named **CMakeLists.txt**, files that are **include'd** should be named ***.cmake**.

Reply:
Fixed